File: backend/src/main/java/com/fashon/domain/entity/Category.java
package com.fashon.domain.entity;

import com.fashon.domain.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Where;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "categories")
@Getter
@Setter
@NoArgsConstructor
@Where(clause = "deleted_at IS NULL")
public class Category extends BaseEntity {

    @Column(name = "name", nullable = false, columnDefinition = "NVARCHAR(255)")
    private String name;

    @Column(name = "slug", nullable = false, unique = true, columnDefinition = "NVARCHAR(255)")
    private String slug;

    @Column(name = "description", columnDefinition = "NVARCHAR(1000)")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private Category parent;

    @OneToMany(mappedBy = "parent", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Category> children = new ArrayList<>();

    @Column(name = "sort_order")
    private Integer sortOrder = 0;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;

    public Category(String name, String slug, String description) {
        this.name = name;
        this.slug = slug;
        this.description = description;
    }

    public void addChild(Category child) {
        children.add(child);
        child.setParent(this);
    }

    public void removeChild(Category child) {
        children.remove(child);
        child.setParent(null);
    }
}

File: frontend/src/types/category.ts
export interface CategoryDTO {
    id: number;
    name: string;
    slug: string;
    description: string | null;
    parentId: number | null;
    parentName: string | null;
    sortOrder: number;
    isActive: boolean;
    children: CategoryDTO[];
}

export interface CreateCategoryRequest {
    name: string;
    slug: string;
    parentId?: number | null;
    description?: string;
    sortOrder?: number;
    isActive?: boolean;
}

export interface UpdateCategoryRequest {
    name?: string;
    slug?: string;
    parentId?: number | null;
    description?: string;
    sortOrder?: number;
    isActive?: boolean;
}
